_drafts/hw_actors.md

```diff
@@ -23,28 +26,26 @@ This is why concepts where interrupts are used as the main 'execution engine' ar
 
 
 ## The concept
 
-
 The classic actor model describes an actor as a message handler, similar to an interrupt handler. When handling messages, the actor may:
 
 - create new actors
 - send messages to other actors
-- read/write its own local state
+- read/write its local state
 
-Actors use messages and queues to communicate. Since interrupt controller has no such functionality, it should be implemented in software. So, there are only three types of software objects: queues, actors and messages.
+Actors use messages and queues to communicate. Since the interrupt controller has no such functionality, it should be implemented in software. So, there are only three types of software objects: queues, actors, and messages.
 
-Despite the actor and its mailbox being tightly coupled in classic models, I think it is less practical than relying on separate queues. The latter case is more flexible, an actor may choose which queue to poll depending on the message being processed. Here, it is assumed that the actor may only have one message, and it has to poll some queue every time it finishes handling the current message.
+Despite the actor and its mailbox being tightly coupled in classic models, it is less practical than relying on separate queues. The latter case is more flexible. Depending on the message being processed, an actor may choose which queue to poll. Here, it is assumed that the actor may only have one message, and it has to poll some queue every time it finishes handling the current message.
 
-All available interrupt vectors are divided into two classes: those that produce messages and those that consume messages. It is expected that interrupts designated to peripheral devices will post messages to queues that in turn have subscribed actors. 
+All available interrupt vectors are divided into two classes: those producing and consuming messages. Interrupts designated to peripheral devices are expected to post messages to queues that, in turn, have subscribed actors.
 
-Let's implement simplest 'blinky' application with one actor who handles messages from systick interrupt and controls onboard LED. The full source of the example for STM32 Bluepill board may be found [here](/example/hw-actors/main.c). The messages have to be sent from systick interrupt to the actor controlling the LED. Therefore, message payload is the next LED state:
+Let's implement the simplest 'blinky' application with one actor who handles messages from systick interrupt and controls onboard LED. The whole source of the example for the STM32 Bluepill board may be found [here](/example/hw-actors/main.c). The messages must be sent from systick interrupt to the actor controlling the LED. Therefore, the message payload is the next LED state:
 
         ```c
```

Comment:
Looks like markdown doesn't like the indenting here so the syntax highlighting isn't coming through. Just need to left-justify and it works 👍